src/utils/eip712/lexer.rs

```diff
@@ -0,0 +1,80 @@
+//!
+//!  Lookup table layout
+//!  ===================
+//!
+//!  ```text
+//!  EOF    ;      :      ,      .      (      )      {      }      [      ]      =>
+//!  IDENT  BLTIN  CONTR  LIB    IFACE  ENUM   STRUCT MODIF  EVENT  FUNCT  VAR    ANON
+//!  AS     ASM    BREAK  CONST  CONTIN DO     DELETE ELSE   EXTERN FOR    HEX    IF
+//!  INDEX  INTERN IMPORT IS     MAP    MEM    NEW    PAY    PULIC  PRAGMA PRIV   PURE
+//!  RET    RETNS  STORAG SUPER  THIS   THROW  USING  VIEW   WHILE  RESERV T_BOOL T_ADDR
+//!  T_STR  T_BYT  T_BYTS T_INT  T_UINT T_FIX  T_UFIX L_TRUE L_FALS L_HEX  L_INT  L_RAT
+//!  L_STR  E_ETH  E_FINN E_SZAB E_WEI  T_YEAR T_WEEK T_DAYS T_HOUR T_MIN  T_SEC  :=
+//!  =:     ++     --     !      ~      *      /      %      **     +      -      <<
+//!  >>     <      <=     >      >=     ==     !=     &      ^      |      &&     ||
+//!  ?      =      +=     -=     *=     /=     %=     <<=    >>=    &=     ^=     |=
+//!  ERRTOK ERREOF
+//!  ```
+//!
+
+use logos::{Lexer, Logos};
+#[derive(Default, Clone, Copy)]
+pub struct TypeSize(pub u8, pub u8);
+
+#[derive(Debug, PartialEq, Clone, Copy, Logos)]
+#[logos(extras = TypeSize)]
+pub enum Token {
+    #[regex("[a-zA-Z_$][a-zA-Z0-9_$]*")]
+    Identifier,
+
+    #[regex("bytes1|bytes[1-2][0-9]?|bytes3[0-2]?|bytes[4-9]", validate_bytes)]
```

Comment:
Now you have `bytes1` covered twice. But this makes lexer a bit more complex when it matches this one twice.